packages/iTwinUI-react/src/core/Table/Table.tsx

```diff
@@ -659,6 +663,10 @@ export const Table = <
     setPageSize(pageSize);
   }, [pageSize, setPageSize]);
 
+  const numSelectedRows =
+    selectionMode === 'single'
+      ? 0
+      : rows.filter((row) => row.isSelected).length;
```

Comment:
I'm not sure because I'm not the most confident with using hooks, but do we usually wrap such things with `useMemo` or `useCallback`, etc.? Not sure

```suggestion
  const numSelectedRows = React.useMemo(
    () =>
      selectionMode === 'single'
        ? 0
        : rows.filter((row) => row.isSelected).length,
    [rows, selectionMode],
  );
```

Reply:
Yes, this should be wrapped with `useMemo` because it is a variable that changes only when `rows` or `selectionMode` changes. We use `useCallback` for function calls and in most cases we do that in order to prevent component that is getting that function from rerendering unnecessary.

Reply:
I have moved this count under `paginatorRendererProps` and removed the `numSelectedRows` variable